src/pmc_simulation.jl

```diff
@@ -21,42 +21,57 @@ mutable struct PMCSimulation
 end
 
 function _set_up_starting_vectors(
-    start_at, n_replicas, n_spectral, style, initiator, threading, copy_vectors
+    ham, start_at, n_replicas, n_spectral, style, initiator, threading, copy_vectors
 )
     if start_at isa AbstractMatrix && size(start_at) == (n_replicas, n_spectral)
         if eltype(start_at) <: Union{AbstractDVec, RMPI.MPIData} # already dvecs
             if copy_vectors
-                return SMatrix{n_spectral, n_replicas}(deepcopy(v) for v in start_at)
+                return SMatrix{n_replicas, n_spectral}(deepcopy(v) for v in start_at)
             else
-                return SMatrix{n_spectral, n_replicas}(v for v in start_at)
+                return SMatrix{n_replicas, n_spectral}(v for v in start_at)
             end
         else
-            return SMatrix{n_spectral, n_replicas}(
+            return SMatrix{n_replicas, n_spectral}(
                 default_starting_vector(sa; style, initiator, threading) for sa in start_at
             )
         end
+    elseif n_spectral > 1
+        basis = build_basis(ham; cutoff=diagonal_element(ham, starting_address(ham)) + 5)
+        basis = sort!(basis; by=x -> diagonal_element(ham, x))[1:2*n_spectral]
+        trunc = BasisSetRepresentation(ham, basis; filter=Returns(false), sizelim=Inf)
+        vecs = eigvecs(Matrix(trunc))
+        if threading
+            eigs = [PDVec(zip(basis, 10*vecs[:,i]); style, initiator) for i in 1:n_spectral]
+        else
+            if initiator == NonInitiator()
+                eigs = [DVec(zip(basis, 10*vecs[:,i]); style) for i in 1:n_spectral]
+            else
+                eigs = [InitiatorDVec(zip(basis,10*vecs[:,i]);style,initiator) for i in 1:n_spectral]
+            end
+        end
```

Comment:
I like this idea a lot, but we should be careful. It's possible to setup a Hamiltonian that's huge, but still has all diagonal elements below `diagonal_element(ham, starting_address(ham)) + 5`. Running this on it will not work.

Reply:
As discussed, I have the same concern. I would suggest to not use an energy cutoff, but instead implement a new option for `build_basis` where you specify a lower bound, e.g. called `mininum_size`, that allows `build_basis` to stop generating new addresses once the `mininum_size` has been reached. 

This will require modifying the code of `build_basis`, but should be fairly easy to implement.

Reply:
I've tried to implement this and it seems to work for me, but one of the tests failed and I'm not understanding what happened - could somebody please take a look?

Reply:
The failing test has nothing to do with your (or our) code. A different package which is a dependency of our code (here `REPL.jl`) failed to precompile on the nightly version of Julia. This can happen when things are changed in the bleeding-edge (and unreleased) development version of Julia. Usually these problems will disappear by themselves.